### ISSUE TITLE
homebridge-hue v.0.14.x is breaking Philips Hue Tap switches

### ISSUE THREAD

**Issue by jostrasser** [ebaauw/homebridge-hue#1192]
#### Issue
After updating to v0.14.x all my Philips Hue Tap switches are broken.
They were still visible in HomeKit, but could no longer be control lamps.
Downgrading to v0.13.70 fixed the issue.

No errors seen in the homebridge.log.
Pressing buttons on the switches are mirrored to the Philips Hue app but not translated to HomeKit via homebridge-hue.

#### Debug Files
[homebridge-hue.json.gz](https://github.com/user-attachments/files/17481043/homebridge-hue.json.gz)

Switch example:

![image](https://github.com/user-attachments/assets/c0073ba4-0c68-4640-b0c6-a2485f8955a4)

**Comment by ebaauw** [comment]
My bad.  In the refactor I kept only the values for `state/buttonevent` that the Hue API v1 reports (the raw ZGP commands 24, 17, 18, and 19).  However, I forgot that the Hue API v2 event stream uses values 1000, 2000, 3000, and 4000 instead.  To make matters worse, Homebridge Hue has been ignoring the v1 reported values since v0.13.57, see #1148.

Looking at the code, the same issue exists for ZGP Friends-of-Hue switches (albeit with different values).

You should see debug messages in the Homebridge log (when running Homebridge in DEBUG mode), that Homebridge Hue sees the button events:
```
[10/22/2024, 10:53:06 PM] [Hue] Hue tap switch 1: state changed event: {"buttonevent":1000,"lastupdated":"2024-10-22T20:53:06.966"}
[10/22/2024, 10:53:06 PM] [Hue] Hue tap switch 1: sensor buttonevent 1000 on 2024-10-22T20:53:06.966
```
But the message that the event is forwarded to HomeKit is missing:
```
[10/22/2024, 10:53:06 PM] [Hue] Hue tap switch 1 1: homekit button single press
```

Luckily, the fix is easy enough.

**Comment by ebaauw** [comment]
Fix in beta v0.14.2-0.

I tested the Hue tap, a FoH switch, the Hue tap dial, and the Lutron Aurora.  They now seem to work.

I see I am still exposing buttons 5 and 6 on the Hue tap and the FoH switches (for pressing two buttons simultaneously), but these events aren't reported by the Hue API v2 event stream.  So the buttons will never be actived.  Maybe I should remove these?

**Comment by jostrasser** [comment]
Thanks for the quick fix @ebaauw !

Yes, I only see 4 buttons exposed, each with a short and long press. 

I never used two buttons simultaneously, they are not configured on my side. For me they can be removed.

If you remove 5 & 6, will it require a reconfiguration of the switches or will only the two buttons disappear?

Thanks JO!

**Comment by ebaauw** [comment]
Released v0.14.2.

The combi buttons 5 and 6 still work with the v1 (round) Hue bridge and with a v2 (square) Hue bridge with ancient firmware (from before the introduction of the event stream), so I'll leave them in for now.  They will not be exposed by Homebridge Hue2.

**Comment by jostrasser** [comment]
Good to know!

No issues so far with the new release! :) thank you!
